File: Projects/MultiFileReader/modules/yml_parser_test.py
#
# Nimish Nayak
# 10-06-2017
#

#
# Yml Parser - Class Tests   
#

from yml_parser import YmlParser
import os

test_file = os.path.join(os.getcwd(),"Test.yml")

def test_yml_parser_with_missing_values():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("""users:
  - name: Paul
    active: false
    value: 100
  - name: Ben
    active: true""")

    y = YmlParser(test_file)
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

def test_yml_parser_with_malformed_yml():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("""users:
  - name Paul
    active  false
    value 100
  - name: Ben
    active: true""")

    y = YmlParser(test_file)
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

def test_yml_parser_with_blank_file():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("")

    y = YmlParser(test_file)
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

def test_yml_parser_with_just_headers():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("users:")

    y = YmlParser(test_file)
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

    # delete the test file as not required 
    os.remove(test_file)

File: Projects/MultiFileReader/modules/yml_parser_test.php
<?php

#
# Nimish Nayak
# 05-07-2017
#

# Test Project to test the code base  
# Functional Test cases

# Note: Developed using PHP 7.0.18-0ubuntu0.16.04.1 (cli) ( NTS )

require_once("./modules/yml_parser.php");
require_once("./modules/utils.php");

class YamlParserTest extends PHPUnit_Framework_TestCase
{
    # Tests 

    public function testParser() {
        
        $testFile1 = Utils::joinPaths(getcwd(),"testdata","test_correct.yml"); 
        $this->yamlParser = new YmlParser($testFile1);
        # get the list of dictinary of the yml contents 
        $this->yamlParser->parseFile();
        # process the value 
        $this->yamlParser->processFile();
        # get the processed value from the parser as a string
        $this->data = $this->yamlParser->getValue();              
        $this->assertEquals($this->data, "900");

    }

    public function testParserWithIncorrectYaml() {
        
        $testFile2 = Utils::joinPaths(getcwd(),"testdata","test_incorrect.yml"); 
        $this->yamlParser = new YmlParser($testFile2);
        # get the list of dictinary of the yml contents 
        $this->yamlParser->parseFile();
        # process the value 
        $this->yamlParser->processFile();
        # get the processed value from the parser as a string
        $this->data = $this->yamlParser->getValue();              
        $this->assertEquals($this->data, 0);
    }

}

?>